public/components/register_model/model_source.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useMemo, useCallback } from 'react';
+import {
+  EuiSpacer,
+  EuiText,
+  EuiComboBoxOptionOption,
+  EuiLink,
+  EuiFormRow,
+  EuiComboBox,
+} from '@elastic/eui';
+
+import { useController, useFormContext } from 'react-hook-form';
+import { useMonitoring } from '../monitoring/use_monitoring';
+
+export const ModelSource = () => {
+  const { allExternalConnectors } = useMonitoring();
```

Comment:
`useMonitoring` has many other logic, how about separate another react hook to fetch all external connectors?